plugin/core/protocol.py

```diff
@@ -6133,3 +6133,5 @@ def to_lsp(self) -> 'Position':
 
 # Temporary for backward compatibility with LSP packages.
 RangeLsp = Range
+CompletionItemDefaults = __ItemDefaults_Type_15
+CompletionEditRange = __EditRange_Type_15
```

Comment:
this is not great, but I need those types

I avoided importing `__ItemDefaults_Type_15` directly, instead I declared `CompletionItemDefaults`, so if the generated type changes I just need to update this place... but It would be good to not deal with this at all.

Any suggestions? 

I will try to find a `Pick` typescript alternative in mypy.

Reply:
It would need to be fixed in the types generator but how exactly I'm not sure. First thing that comes to mind is that it's name (or suffix) should be based on **where it's used** (in which parent object) since then the name shouldn't change based on other types being added earlier.

Reply:
I tired doing that before, 
but run into some issue like the generated underscore types could have clashing names.
But I will try again.

Reply:
Let me just know if that is a blocker for this PR, or something that could be done later?

Reply:
Shouldn't be a blocker since it won't change on us by itself.

Reply:
I will work on it now.

Reply:
8ca4d8e